File: frontend/src/sentry.ts
import * as Sentry from "@sentry/react";
import { browserTracingIntegration, replayIntegration } from "@sentry/react";

//TODO: Move is somewhere else
declare global {
    interface Window {
        __APP_CONFIG__?: {
            APP_VERSION?: string;
            SENTRY_DSN?: string;
            ENVIRONMENT?: string;
            SENTRY_TRACES_SAMPLE_RATE?: number;
            SENTRY_REPLAYS_SESSION_SAMPLE_RATE?: number;
            SENTRY_REPLAYS_ON_ERROR_SAMPLE_RATE?: number;
            SENTRY_SEND_DEFAULT_PII?: boolean;
            SENTRY_ENABLE_LOGS?: boolean;
            API_URL?: string;
        };
    }
}



export function initSentry() {
    const cfg = window.__APP_CONFIG__;
    const dsn = cfg?.SENTRY_DSN || import.meta.env.VITE_SENTRY_DSN;
    const release = import.meta.env.VITE_VERSION;

    // Check for empty or missing DSN
    if (!dsn || dsn.trim() === '') {
        console.log('[Sentry] Initialization skipped - no DSN configured');
        return;
    }

    console.log('[Sentry] Initializing with release:', release);

    Sentry.init({
        dsn,
        release,
        integrations: [
            browserTracingIntegration(),
            replayIntegration()
        ],
        tracesSampleRate: cfg?.SENTRY_TRACES_SAMPLE_RATE || import.meta.env.VITE_SENTRY_TRACES_SAMPLE_RATE,
        replaysSessionSampleRate: cfg?.SENTRY_REPLAYS_SESSION_SAMPLE_RATE || import.meta.env.VITE_SENTRY_REPLAYS_SESSION_SAMPLE_RATE,
        replaysOnErrorSampleRate: cfg?.SENTRY_REPLAYS_ON_ERROR_SAMPLE_RATE || import.meta.env.VITE_SENTRY_REPLAYS_ON_ERROR_SAMPLE_RATE,
        environment: cfg?.ENVIRONMENT || import.meta.env.VITE_SENTRY_ENVIRONMENT,
        sendDefaultPii: cfg?.SENTRY_SEND_DEFAULT_PII || import.meta.env.VITE_SENTRY_SEND_DEFAULT_PII,
        enableLogs: cfg?.SENTRY_ENABLE_LOGS || import.meta.env.VITE_SENTRY_ENABLE_LOGS,
    });
}

File: backend/watercrawl/sentry.py
from sentry_sdk import init as sentry_sdk_init


def init_sentry():
    from . import __version__
    from django.conf import settings

    if not settings.SENTRY_DSN:
        return

    sentry_sdk_init(
        dsn=settings.SENTRY_DSN,
        traces_sample_rate=settings.SENTRY_TRACES_SAMPLE_RATE,
        send_default_pii=settings.SENTRY_SEND_DEFAULT_PII,
        enable_logs=settings.SENTRY_ENABLE_LOGS,
        release=__version__,
        http_proxy=settings.SENTRY_HTTP_PROXY or None,
        https_proxy=settings.SENTRY_HTTPS_PROXY or None,
        environment=settings.SENTRY_ENVIRONMENT,
        debug=settings.SENTRY_DEBUG,
    )
